core/src/main/java/org/springframework/security/authorization/method/AuthorizationAnnotationUtils.java

```diff
@@ -50,6 +56,36 @@
  */
 final class AuthorizationAnnotationUtils {
 
+	static <A extends Annotation> Function<AnnotatedElement, A> withPlaceholderResolver(Class<A> type) {
+		Function<MergedAnnotation<A>, A> map = (mergedAnnotation) -> {
+			A annotation = mergedAnnotation.synthesize();
+			if (mergedAnnotation.getMetaSource() == null) {
+				return annotation;
+			}
+			String expression = (String) mergedAnnotation.asMap().get("value");
+			MutablePropertySources propertySources = new MutablePropertySources();
+			Map<String, Object> annotationProperties = mergedAnnotation.getMetaSource().asMap();
+			propertySources.addFirst(new MapPropertySource("annotation", annotationProperties));
+			PropertySourcesPropertyResolver propertyResolver = new PropertySourcesPropertyResolver(propertySources);
+			AnnotatedElement annotatedElement = (AnnotatedElement) mergedAnnotation.getSource();
+			String value = propertyResolver.resolvePlaceholders(expression);
+			return MergedAnnotation.of(annotatedElement, type, Collections.singletonMap("value", value)).synthesize();
```

Comment:
I've only glanced at this, but I'm wondering if it might be easier to work with `PropertyPlaceholderHelper` instead.

For example, see the `StaticStringValueResolver` in `spring-test`:

https://github.com/spring-projects/spring-framework/blob/00577ed80acfc22104b41d6a631f433a3fa7e6af/spring-test/src/main/java/org/springframework/test/web/servlet/setup/StandaloneMockMvcBuilder.java#L582-L597